### ISSUE TITLE
Memory leak?

### ISSUE THREAD

**Issue by rainwoodman** [mpip/pfft#24]
Do you run valgrind on pfft to check for leaks?

**Comment by mpip** [comment]
Yes, I'm currently looking for some existing memory leaks in PFFT with 
valgrind. However, there are a lot of false positives due to MPI.

Am 10.06.2016 um 00:31 schrieb Yu Feng:

**Comment by alanvbs** [comment]
Maybe this information can be  useful.  
Compiling the mpich library (I have tested this with the version 3.2)  with these configure options:
`/configure --prefix=PATH --enable-shared --enable-threads=multiple --enable-g=all --enable-error-checking=all --enable-error-messages=all`
I am being able to use the valgrind  and also  the `-fsanitize=address -fsanitize=leak` options from newer versions of the gcc without false alarms regarding the MPI library (I have tested these options only in personal projects. I don't known if some MPI functions can still trigger false alarms). To use the option `-fsanitize=threads` (useful for detecting data races) I had to compile the gcc with this  configure option `--disable-linux-futex`.

It somewhat cumbersome to setup but for me was very useful, specially since using the fsanitize options is way faster than using the valgrind.

**Comment by mpip** [comment]
Thank you very much. I will give it a try.

-------- Ursprüngliche Nachricht --------
Von: alanvbs <notifications@github.com> 
Datum:  
An: mpip/pfft <pfft@noreply.github.com> 
Cc: Michael Pippig <michael.pippig@mathematik.tu-chemnitz.de>,Comment <comment@noreply.github.com> 
Betreff: Re: [mpip/pfft] Memory leak? (#24) 
 
Maybe this information can be useful.

Compiling the mpich library (I have tested this with the version 3.2) with these configure options:
/configure --prefix=PATH --enable-shared --enable-threads=multiple --enable-g=all --enable-error-checking=all --enable-error-messages=all
I am being able to use the valgrind and also the -fsanitize=address -fsanitize=leak options from newer versions of the gcc without false alarms regarding the MPI library (I have tested these options only in personal projects. I don't known if some MPI functions can still trigger false alarms). To use the option -fsanitize=threads (useful for detecting data races) I had to compile the gcc with this configure option --disable-linux-futex.

It somewhat cumbersome to setup but for me was very useful, specially since using the fsanitize options is way faster than using the valgrind.
